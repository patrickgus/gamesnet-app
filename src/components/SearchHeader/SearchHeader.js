import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./SearchHeader.css";

export default class SearchHeader extends Component {
  render() {
    return (
      <header className="SearchHeader">
        <form className="searchBarNav">
          <label htmlFor="search">Search: </label>
          <input type="text" id="search" name="search" />

          <label htmlFor="sort">Sort: </label>
          <select id="sort" name="sort">
            <option value="title">Title</option>
            <option value="rating">Rating</option>
          </select>
          <button type="submit">Go</button>
        </form>
        <Link to={"/addgame"}>Add a Game</Link>
      </header>
    );
  }
}
