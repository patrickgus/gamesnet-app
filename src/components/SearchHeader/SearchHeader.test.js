import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import SearchHeader from "./SearchHeader";

describe("SearchHeader component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <SearchHeader />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SearchHeader />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
