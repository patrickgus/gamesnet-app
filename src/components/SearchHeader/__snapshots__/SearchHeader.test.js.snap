// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader component renders the UI as expected 1`] = `
<header
  className="SearchHeader"
>
  <form
    className="searchBarNav"
  >
    <label
      htmlFor="search"
    >
      Search: 
    </label>
    <input
      id="search"
      name="search"
      type="text"
    />
    <label
      htmlFor="sort"
    >
      Sort: 
    </label>
    <select
      id="sort"
      name="sort"
    >
      <option
        value="title"
      >
        Title
      </option>
      <option
        value="rating"
      >
        Rating
      </option>
    </select>
    <button
      type="submit"
    >
      Go
    </button>
  </form>
  <a
    href="/addgame"
    onClick={[Function]}
  >
    Add a Game
  </a>
</header>
`;
